src/core/server/connector/connector_manager.cc

```diff
@@ -0,0 +1,43 @@
+#include "core/server/connector/connector_manager.h"
+
+#include <thread>
+#include <vector>
+
+#include <gflags/gflags.h>
+#include <glog/logging.h>
+
+#include "core/frame_response.h"
+#include "core/server/connector/connector.h"
+#include "core/player.h"
+
+using namespace std;
+
+DEFINE_bool(realtime, true, "use realtime");
+DEFINE_bool(no_timeout, false, "if true, wait ai's thought without timeout");
+
+ConnectorManager::ConnectorManager(unique_ptr<Connector> p1, unique_ptr<Connector> p2, bool timeout)
+    : connectors_{ move(p1), move(p2) }
+    , waitTimeout_(timeout)
+{
+}
+
+bool ConnectorManager::receive(int frameId, vector<FrameResponse> cfr[NUM_PLAYERS])
+{
+    bool connection = true;
+    thread threads[NUM_PLAYERS];
```

Comment:
Hmm, making 2 threads for each receive() ??
This might cause unacceptable overhead (maybe 2+ milliseconds)
It's good if we invoke receiver threads when connector manager is starting, and it puts FrameResponse to a blocking queue. receive() receives a FrameResponse from the blocking queue, and it also might wait for timeout.